src/psyclone/tests/psyir/backend/fortran_test.py

```diff
@@ -552,10 +552,13 @@ def test_fw_kernelschedule(fort_writer, monkeypatch):
 # within previous tests
 
 
-def test_fw_binaryoperator(fort_writer):
+@pytest.mark.parametrize("binary_intrinsic", ["mod", "matmul", "max", "min",
```

Comment:
Apologies but I've just realised that `matmul` isn't valid in this test either - please could you separate it out.

Reply:
I'm guessing you mean matmul(1.0, 1.0). Your suggestion of the addition of a compile test also picked this up :-)